BankApplicationPractice-main/BankLibrary/Bank.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace BankLibrary
+{
+    public class Bank<T> where T : Account
+    {
+        private readonly List<T> _accounts = new();
+
+        public void OpenAccount(OpenAccountParameters parameters)
+        {
+            if (GetType().GetGenericArguments()[0] == typeof(DepositAccount))
```

Comment:
check if this will work: `if (typeof(T) == typeof(DepositAccount))`

Reply:
checked, its working. ty